File: system-test/dotnet/SystemTests/Legacy/Mod04/Base/BaseClientTest.cs
using SystemTests.TestInfrastructure.Configuration;
using Dsl.Core;
using Driver.Adapter.External.Erp.Client;
using Driver.Adapter.Api.Client;
using Driver.Adapter.Ui.Client;
using Driver.Adapter.External.Tax.Client;
using Xunit;

namespace SystemTests.Legacy.Mod04.Base;

public abstract class BaseClientTest : BaseConfigurableTest, IAsyncLifetime
{
    protected readonly Dsl.Core.Configuration _configuration;

    protected MyShopUiClient? _shopUiClient;
    protected MyShopApiClient? _shopApiClient;

    protected ErpRealClient? _erpClient;
    protected TaxRealClient? _taxClient;

    protected BaseClientTest()
    {
        _configuration = LoadConfiguration();
    }

    public virtual Task InitializeAsync()
    {
        return Task.CompletedTask;
    }

    protected async Task SetUpMyShopUiClientAsync()
    {
        _shopUiClient = await MyShopUiClient.CreateAsync(_configuration.MyShopUiBaseUrl);
    }

    protected void SetUpMyShopApiClient()
    {
        _shopApiClient = new MyShopApiClient(_configuration.MyShopApiBaseUrl);
    }

    protected void SetUpExternalClients()
    {
        _erpClient = new ErpRealClient(_configuration.ErpBaseUrl);
        _taxClient = new TaxRealClient(_configuration.TaxBaseUrl);
    }

    public virtual async Task DisposeAsync()
    {
        if (_shopUiClient != null)
            await _shopUiClient.DisposeAsync();
        _shopApiClient?.Dispose();
        _erpClient?.Dispose();
        _taxClient?.Dispose();
    }
}

File: system-test/java/src/test/java/com/mycompany/myshop/systemtest/legacy/mod04/base/BaseClientTest.java
package com.mycompany.myshop.systemtest.legacy.mod04.base;

import com.mycompany.myshop.systemtest.configuration.BaseConfigurableTest;
import com.mycompany.myshop.systemtest.configuration.Configuration;
import com.mycompany.myshop.testkit.driver.adapter.external.erp.client.ErpRealClient;
import com.mycompany.myshop.testkit.driver.adapter.external.tax.client.TaxRealClient;
import com.mycompany.myshop.testkit.driver.adapter.api.client.MyShopApiClient;
import com.mycompany.myshop.testkit.driver.adapter.ui.client.MyShopUiClient;
import com.mycompany.myshop.systemtest.infrastructure.playwright.BrowserLifecycleExtension;
import com.mycompany.myshop.testkit.common.Closer;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;

public abstract class BaseClientTest extends BaseConfigurableTest {
    protected Configuration configuration;

    protected MyShopUiClient myShopUiClient;
    protected MyShopApiClient myShopApiClient;
    protected ErpRealClient erpClient;
    protected TaxRealClient taxClient;

    @BeforeEach
    protected void setUpConfiguration() {
        configuration = loadConfiguration();
    }

    protected void setUpMyShopUiClient() {
        myShopUiClient = new MyShopUiClient(configuration.getMyShopUiBaseUrl(), BrowserLifecycleExtension.getBrowser());
    }

    protected void setUpMyShopApiClient() {
        myShopApiClient = new MyShopApiClient(configuration.getMyShopApiBaseUrl());
    }

    protected void setUpExternalClients() {
        erpClient = new ErpRealClient(configuration.getErpBaseUrl());
        taxClient = new TaxRealClient(configuration.getTaxBaseUrl());
    }

    @AfterEach
    void tearDown() {
        Closer.close(myShopUiClient);
        Closer.close(myShopApiClient);
        Closer.close(erpClient);
        Closer.close(taxClient);
    }

}
